### PR TITLE
airflow: add dag integrity test

### LINKED ISSUE #537: Create DAG Integrity tests
currently were are not checking if our dags are syntatically correct. This should be fixed

### PR DESCRIPTION
**Pull request by DonHaul**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -48,10 +48,10 @@ x-airflow-common: &airflow-common
   # In order to add custom dependencies or upgrade provider packages you can use your extended image.
   # Comment the image line, place your Dockerfile in the directory where you placed the docker-compose.yaml
   # and uncomment the "build" line below, Then run `docker-compose build` to build the images.
-  image: apache/airflow:2.8.3-python3.11
+  # image: apache/airflow:2.9.3-python3.11
```
Comment (drjova) [inline docker-compose.yaml]: we can remove it completely

File: workflows/dags/author/author_update/author_update.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def set_author_update_workflow_status_to_completed(**context):
             collection=AUTHORS,
         )
 
-    @task.branch(provide_context=True)
+    @task.branch()
```
Comment (drjova) [inline workflows/dags/author/author_update/author_update.py:102]: why did  we remove it?

Comment (DonHaul) [reply workflows/dags/author/author_update/author_update.py:102]: It is now implicitly detected 
https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-operators-python-pythonoperator

Even in the code, they pop it.
https://airflow.apache.org/docs/apache-airflow/2.9.3/_modules/airflow/operators/python.html#:~:text=if%20kwargs.get(%22provide_context%22)%3A

Comment (drjova) [reply workflows/dags/author/author_update/author_update.py:102]: Thanks for the detailed explaination.

**Review by drjova — APPROVED** [review]

**Pull request merged**